lib/plugins/aws/deploy/compile/functions/tests/index.js

```diff
@@ -163,9 +163,217 @@ describe('AwsCompileFunctions', () => {
+        },
+        func1: {
+          handler: 'func.function.handler',
+          name: 'new-service-dev-func1',
+          roleArn: 'some:aws:arn:xx1:*:*',
+        },
+      };
+
+      awsCompileFunctions.compileFunctions();
+
+      expect(awsCompileFunctions.serverless.service.provider.compiledCloudFormationTemplate
+        .Resources.Func0LambdaFunction.Properties.Role
+      ).to.deep.equal(awsCompileFunctions.serverless.service.provider.roleArn);
+      expect(awsCompileFunctions.serverless.service.provider.compiledCloudFormationTemplate
+        .Resources.Func1LambdaFunction.Properties.Role
+      ).to.deep.equal(awsCompileFunctions.serverless.service.functions.func1.roleArn);
+    });
+
+    it('should not add the default role and policy if provider has an roleArn', () => {
+      awsCompileFunctions.serverless.service.provider.name = 'aws';
+      awsCompileFunctions.serverless.service.provider.roleArn = 'some:aws:arn:xxx:*:*';
+
+      awsCompileFunctions.compileFunctions();
+
+      // eslint-disable-next-line no-unused-expressions
+      expect(awsCompileFunctions.serverless.service.provider.compiledCloudFormationTemplate
+        .Resources.IamPolicyLambdaExecution
+      ).to.be.undefined;
+      // eslint-disable-next-line no-unused-expressions
+      expect(awsCompileFunctions.serverless.service.provider.compiledCloudFormationTemplate
+        .Resources.IamRoleLambdaExecution
+      ).to.be.undefined;
+    });
+
+    it('should not add the default role and policy if all functions have an roleArn', () => {
+      awsCompileFunctions.serverless.service.provider.name = 'aws';
+      awsCompileFunctions.serverless.service.functions = {
+        func0: {
+          handler: 'func.function.handler',
+          name: 'new-service-dev-func0',
+          roleArn: 'some:aws:arn:xx0:*:*',
+        },
+        func1: {
+          handler: 'func.function.handler',
+          name: 'new-service-dev-func1',
+          roleArn: 'some:aws:arn:xx1:*:*',
+        },
+      };
+
+      awsCompileFunctions.compileFunctions();
+
+      // eslint-disable-next-line no-unused-expressions
+      expect(awsCompileFunctions.serverless.service.provider.compiledCloudFormationTemplate
+        .Resources.IamPolicyLambdaExecution
+      ).to.be.undefined;
+      // eslint-disable-next-line no-unused-expressions
+      expect(awsCompileFunctions.serverless.service.provider.compiledCloudFormationTemplate
+        .Resources.IamRoleLambdaExecution
+      ).to.be.undefined;
+    });
+
+    it('should not add default role/policy if all functions can has roleArn', () => {
+      awsCompileFunctions.serverless.service.provider.name = 'aws';
+      awsCompileFunctions.serverless.service.provider.roleArn = 'some:aws:arn:xxx:*:*';
+      awsCompileFunctions.serverless.service.functions = {
+        func0: {
+          handler: 'func.function.handler',
+          name: 'new-service-dev-func0',
+          // roleArn: 'some:aws:arn:xx0:*:*', // obtain from provider
```

Comment:
the intent was to explicitly indicate that it was purposely not set.  will remove.